integrations/wagmi/src/index.ts

```diff
@@ -39,16 +40,16 @@ export function injectedWithSapphire(): ReturnType<typeof injected> {
             // Note: providers are cached as connectors are frequently retrieved
             //       it prevents sapphire wrapper being called multiple times
             //       which spams the RPC with oasis_callDataPublicKey requests
-            if (!(window.ethereum in cachedProviders)) {
-              cachedProviders[window.ethereum] = wrapEIP1193Provider(
-                window.ethereum,
-              );
+            if( ! (cachedProviders.has(window.ethereum)) ) {
+              const wp = wrapEIP1193Provider(window.ethereum as any) as EIP1193Provider & SapphireAnnex;
```

Comment:
If you want to cast this to `EIP1193Provider & SapphireAnnex`, please do it the following way:
```typescript
const wp = wrapEIP1193Provider(window.ethereum as unknown as EIP1193Provider);
```

Please cast to `unknown` first and then to desired type. `any` messes up the type completely.